src/libraries/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -4251,6 +4251,78 @@ public void Fill_InvalidStartIndexCount_ThrowsArgumentOutOfRangeException(int ar
             AssertExtensions.Throws<ArgumentOutOfRangeException>("count", () => Array.Fill(new string[arrayLength], "", startIndex, count));
         }
 
+        public class Bar
+        {
+            public string Value { get; set; }
+        }
+
+        public class Foo
+        {
+        }
+
+        private static Bar[] CreateBarArray()
+        {
+            return new Bar[]
+            {
+                new Bar() { Value = "0" },
+                new Bar() { Value = "1" },
+                new Bar() { Value = "2" },
+                new Bar() { Value = "3" },
+            };
+        }
+
+        private static int[] CreateInt32Array() => new int[] { 0, 1, 2, 3 };
+
+        [Fact]
+        public static void Fill_Casting()
+        {
+            Bar[] barArray1 = CreateBarArray();
+            Array.Fill<object>(barArray1, new Bar() { Value = "x" });
+            Assert.Equal(Enumerable.Repeat("x", barArray1.Length), barArray1.Select(e => e.Value));
```

Comment:
Consider using `new string[] { 'x', 'x', 'x', 'x' }` here to match the pattern you have in line 4285 below. Removing the call to `Enumerable.Repeat` uncomplicates the test.

(Same comment for line 4289.)

Reply:
I was originally considering parameterised the tests, this would require the second `Fill_Casting` test to be written something like:

https://github.com/dotnet/runtime/blob/8c90687369789013055aedd012970babc1bcb4f1/src/libraries/System.Runtime/tests/System/ArrayTests.cs#L4216-L4223

But this seemed unnecessarily complicated 😋